docroot/profiles/custom/cgov_site/modules/custom/pdq_cancer_information_summary/pdq_cancer_information_summary.module

```diff
@@ -4,3 +4,123 @@
  * @file
  * Contains pdq_cancer_information_summary.module.
  */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function pdq_cancer_information_summary_node_insert(EntityInterface $entity) {
+  if ($entity->bundle() === 'pdq_cancer_information_summary') {
+    _pdq_cis_add_site_sections($entity);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function pdq_cancer_information_summary_node_update(EntityInterface $entity) {
+  if ($entity->bundle() === 'pdq_cancer_information_summary') {
+    _pdq_cis_add_site_sections($entity);
+  }
+}
+
+/**
+ * Common code to add site_section terms.
+ *
+ * @param Drupal\Core\Entity\EntityInterface $entity
+ *   Reference to the PDQ Cancer Information Summary just saved.
+ */
+function _pdq_cis_add_site_sections(EntityInterface $entity) {
+
+  // Collect the values we need from the summary.
```

Comment:
Does the transitioning a summary through the workflow call this code? We had discussed the want of checking what has changed in order to determine if all the queries are needed.

Reply:
I'll add an optimizing check in the morning.

Reply:
Something like: find the `cgov_site_sections` term with the `computed_path` matching the pdq url field, and if its `name` field (since we're not setting the nav label field) matches the short title of the summary, there's nothing to do in hook. How's that?

One last thing to check: I mimicked what I saw in the sample yaml site sections terms (see `000_nav_root-L1.content.yml`), setting the nav label field for the Spanish root term, but not for the English root term. Just want to confirm that this is correct (for some undocumented reason).

Reply:
Another approach I could to take to optimizing away one of the passes through the login in these hooks would be to look at the `status` of the newly saved revision (or its workflow state, which would amount to the same thing), and only create the site sections when the revision was published (in other words, delay until the second revision is created). There's logical incentive for doing it this way, because if in a strange edge case some super-admin came along and edited a summary directly (which the system allows), saving the changes as a draft, it would be more appropriate for the term to match the revision of the summary which is published, particularly since the terms aren't versioned. I thought about doing it this way earlier, but decided instead to follow the logic as written in the ticket for the import/replacement of the summary. Do you have a preference for how I do the optimization?

Reply:
`$entity->original` will give you the previous version of the entity. (For your status or moderation_state check)

We had sketched something on the white board around this approach, and I do not recall what the outcome was. Do you happen to remember?

Reply:
I just pushed a version which does both checks (publishing status of the summary revision and confirmation that the term is already present, with the label unchanged). Using `$entity->original` won't work with this approach, because I'm deferring doing anything with the site sections until I've got public changes to the summary, so the original and current won't be different when it's time to do the work of creating the site sections.

Reply:
A reminder that there's still a minor question or two about the navigation label field hanging above.

Reply:
A more likely scenario than the one described above in which a super-admin edits a summary directly, saving the changes as a draft, would be a publishing job which fails before all of the summaries have been released from draft to published status. I don't think, based on history, that we can afford to dismiss that as an edge case. :-)

Reply:
Ah, this is why we run the tests. 😃 I need to modify the tests to defer looking for the site section until after the release to publishing. In fact, I'm thinking I should probably expand the test to do another save with changes to the summary's short title, and verify that the site section doesn't get updated until after that change has been published.